### PR TITLE
the little construction pr

### PR DESCRIPTION
**Pull request by Varlaisvea**
this was going to be a big construction pr until i realised just how much effort it'd be.
so now it's a little construction pr instead

- you can now make electrochromic window buttons (circuit from autolathe, frame on wall, like fire alarms)
- electrochromic (polarized) window multitool interactions actually work now
- certain floor tiles (primarily metals) can now be welded back into their base forms
- catwalk construction now only eats your metal rods when the catwalk is made
- proof of concept for further tweaks, you can now make those pretty bushes out of grass

honestly nothing too big but it's comfy

### CODE REVIEW

File: code/game/objects/items/stacks/tiles/tile_types.dm [hunk 1]
```diff
@@ -20,9 +20,50 @@
 	drop_sound = 'sound/items/drop/axe.ogg'
 	pickup_sound = 'sound/items/pickup/axe.ogg'
 
+//crafting / welding vars
+	var/datum/material/material //*sigh* i guess this is how we're doing this.
```
Comment (Atermonera) [inline code/game/objects/items/stacks/tiles/tile_types.dm:24]: You _could_ make the tiles a subtype of stack/material, but that almost seems like more work...
Someday, we'll have material on /obj/item (Or maybe even /obj???)

File: code/game/objects/items/stacks/tiles/tile_types.dm [hunk 2]
```diff
@@ -20,9 +20,50 @@
...
+		material = get_material_by_name("[default_type]")
+		if(!material)
+			return INITIALIZE_HINT_QDEL
+		if(material) //sanity check
+			recipes = material.get_recipes()
+			stacktype = material.stack_type
+
+/obj/item/stack/tile/attackby(obj/item/W as obj, mob/user as mob)
+	..()
+	if (istype(W, /obj/item/weldingtool))
```
Comment (Atermonera) [inline code/game/objects/items/stacks/tiles/tile_types.dm:45]:
```suggestion
	if (istype(W, /obj/item/weldingtool))
		// do the thing
		return TRUE
	return ..()
```

File: code/game/objects/items/stacks/tiles/tile_types.dm [hunk 3]
```diff
@@ -20,9 +20,50 @@
...
+		var/obj/item/weldingtool/WT = W
+
+		if(can_weld == FALSE)
+			to_chat("You can't reform these into their original components.")
```
Comment (Atermonera) [inline code/game/objects/items/stacks/tiles/tile_types.dm:49]: Missing return

File: code/game/objects/items/stacks/tiles/tile_types.dm [hunk 4]
```diff
@@ -20,9 +20,50 @@
...
+
+		if(get_amount() < 4)
+			to_chat(user, "<span class='warning'>You need at least four tiles to do this.</span>")
+			return
+
+		if(WT.remove_fuel(0,user))
+			new welds_into(usr.loc)
+			usr.update_icon()
+			for (var/mob/M in viewers(src))
+				M.show_message("<span class='notice'>[src] is shaped by [user.name] with the welding tool.</span>", 3, "<span class='notice'>You hear welding.</span>", 2)
```
Comment (Atermonera) [inline code/game/objects/items/stacks/tiles/tile_types.dm:59]: `visible_message()`

File: code/game/objects/structures/lattice.dm [hunk 5]
```diff
@@ -71,9 +71,12 @@
 		return
 	if (istype(C, /obj/item/stack/rods))
 		var/obj/item/stack/rods/R = C
-		if(R.use(2))
+		if(R.get_amount() < 2)
+			to_chat(user, "<span class='notice'>You need at least two rods to form a catwalk here.</span>")
```
Comment (Atermonera) [inline code/game/objects/structures/lattice.dm:75]: That's a flimsy catwalk \s

**Review by Atermonera — CHANGES_REQUESTED** [review]

File: code/game/objects/items/stacks/tiles/tile_types.dm (hunk 1 above)
Comment (Atermonera) [reply code/game/objects/items/stacks/tiles/tile_types.dm:24]: (Doesn't need a change)

Comment (Varlaisvea) [reply code/game/objects/items/stacks/tiles/tile_types.dm:24]: considered it but then also i realised i value my sanity

File: code/game/objects/items/stacks/tiles/tile_types.dm [hunk 6]
```diff
@@ -41,8 +41,10 @@
 			stacktype = material.stack_type
 
 /obj/item/stack/tile/attackby(obj/item/W as obj, mob/user as mob)
-	..()
 	if (istype(W, /obj/item/weldingtool))
+		// do the thing
+		return TRUE
+	return ..()
```
Comment (Atermonera) [inline code/game/objects/items/stacks/tiles/tile_types.dm:47]: Errr....

Comment (Varlaisvea) [reply code/game/objects/items/stacks/tiles/tile_types.dm:47]: spent the last five minutes looking for // do the thing in the code before realising you meant thats where the code went smh

File: code/game/objects/items/stacks/tiles/tile_types.dm (hunk 3 above)
Comment (Varlaisvea) [reply code/game/objects/items/stacks/tiles/tile_types.dm:51]: github refuses to acknowledge this exists now but it totally does

File: code/game/objects/structures/lattice.dm (hunk 5 above)
Comment (Varlaisvea) [reply code/game/objects/structures/lattice.dm:75]: it's like that one squid game bridge, you have to stand on the rods to walk through safely

**Review by Atermonera — APPROVED** [review]

**Pull request merged**